hkm/management/commands/clean_unused_data.py

```diff
@@ -0,0 +1,46 @@
+from django.core.management.base import BaseCommand
+from django.contrib.auth.models import User
+from hkm.models.models import TmpImage, Feedback, ProductOrder
+from datetime import timedelta
+from django.utils import timezone
+
+DEFAULT_DAYS = 365
+
+class Command(BaseCommand):
+    help = "Clean unused data from application."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "-d", "--days", type=int, help="Specifies the time in days for which old data is deleted", default=DEFAULT_DAYS
+        )
+
+    def handle(self, *args, **kwargs):
+        days = kwargs['days']
+        if not days:
+            days = DEFAULT_DAYS
+
+        self.stdout.write("Old data cleaning started! Removing data older than " + str(days) + " days.")
+        counted_date = timezone.now() - timedelta(days=days)
+
+        # Find and delete unused users
+        unused_users = User.objects.filter(last_login__lte=counted_date)
+        unused_users_count = unused_users.count()
+        unused_users.delete()
+
+        # Find and delete old TmpImages, Feedbacks and Orders
+        unused_temps = TmpImage.objects.filter(modified__lte=counted_date)
+        unused_temps_count = unused_temps.count()
+        unused_feedbacks = Feedback.objects.filter(modified__lte=counted_date)
+        unused_feedbacks_count = unused_feedbacks.count()
+        unused_orders = ProductOrder.objects.filter(modified__lte=counted_date)
+        unused_orders_count = unused_orders.count()
```

Comment:
You should add another condition in the filter here so that only anonymous `Feedbacks` and `ProductOrders` are deleted. Something like

`Feedback.objects.filter(user__isnull=True, ...)`

If they have an active `User` associated with them (e.g. `Feedback.user` is not null), it doesn't matter if they are very old. At some point in the future we might want to be able to show the user's sent feedbacks and/or order history, so we want to keep those in the database until the user goes inactive.

The `TmpImage` deletion can be as it is. We're deleting the images from the bucket anyway within 365 days, so we can get rid of the database rows as well even if they have a `User` association.

Reply:
Ahh, totally forgot that it was supposed to delete only stuff where user is `null`. I'll change this.